plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystemMinio.java

```diff
@@ -0,0 +1,161 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.containers.Minio;
+import io.trino.testing.minio.MinioClient;
+import io.trino.util.AutoCloseableCloser;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+
+import static io.trino.hadoop.ConfigurationInstantiator.newEmptyConfiguration;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertTrue;
+
+public class TestTrinoS3FileSystemMinio
+        extends BaseTestTrinoS3FileSystemObjectStorage
+{
+    public static final String MINIO_ACCESS_KEY = "accesskey";
+    public static final String MINIO_SECRET_KEY = "secretkey";
+
+    private final AutoCloseableCloser closer = AutoCloseableCloser.create();
+    private final String bucketName = "trino-ci-test";
+
+    private Minio minio;
+
+    private MinioClient minioClient;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        minio = Minio.builder()
+                .withEnvVars(ImmutableMap.<String, String>builder()
+                        .put("MINIO_ACCESS_KEY", MINIO_ACCESS_KEY)
+                        .put("MINIO_SECRET_KEY", MINIO_SECRET_KEY)
+                        .buildOrThrow())
+                .build();
+        closer.register(minio);
+        minio.start();
+
+        minioClient = closer.register(minio.createMinioClient());
+        minio.createBucket(bucketName);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        closer.close();
+    }
+
+    @Override
+    protected String getBucketName()
+    {
+        return bucketName;
+    }
+
+    @Override
+    protected Configuration s3Configuration()
+    {
+        Configuration config = newEmptyConfiguration();
+        config.set("trino.s3.endpoint", minio.getMinioAddress());
+        config.set("trino.s3.access-key", MINIO_ACCESS_KEY);
+        config.set("trino.s3.secret-key", MINIO_SECRET_KEY);
+        config.set("trino.s3.path-style-access", "true");
+
+        return config;
+    }
+
+    @Test
+    public void testDeleteNonRecursivelyEmptyBucketRoot()
+            throws Exception
+    {
+        String testBucketName = "trino-delete-bucket-root-empty";
```

Comment:
Due to filesystem caches this may need to be randomized

https://github.com/trinodb/trino/blob/af73f3b4834a7d52bd2d15bf34d61d4f871f8672/testing/trino-testing-containers/src/main/java/io/trino/testing/minio/MinioClient.java#L162-L167

Reply:
I opted for creating a `randomBucketSuffix()` method inspired from 

https://github.com/trinodb/trino/blob/f60bc22016b10793518d7d87e13faf797d6df868/testing/trino-testing/src/main/java/io/trino/testing/sql/TestTable.java#L133-L137

Using `UUID.randomUUID()` doesn't work because MinIO limits the lenght of the buckets to be between `3` and `63` characters.